app/src/main/java/ch/heigvd/pro/b04/android/Utils/Persistent.java

```diff
@@ -0,0 +1,44 @@
+package ch.heigvd.pro.b04.android.Utils;
+
+import android.content.Context;
+import android.content.SharedPreferences;
+
+import ch.heigvd.pro.b04.android.R;
+import ch.heigvd.pro.b04.android.Utils.Exceptions.TokenNotSetException;
+
+public final class Persistent {
+    private Persistent() {};
+
+    /**
+     * This function returns the token if it was set previously.
+     * @param context The app context. Needed to access the Shared Preferences
+     * @return The token value
+     * @throws TokenNotSetException is thrown if no token was stored
+     */
+    public static String getStoredTokenOrError(Context context) throws TokenNotSetException {
+        String defValue = "Error";
+        SharedPreferences preferences = context.getSharedPreferences(
```

Comment:
You probably want to use a normal String as the key, as resources can change based on configuration (for instance if a different locale is applied).

Also, why not use the default preferences (available via [`PreferenceManager.getDefaultSharedPreferences(..)`](https://developer.android.com/reference/android/preference/PreferenceManager#getDefaultSharedPreferences(android.content.Context)) ?

Reply:
Well, at first I thought of using it, because my [main reference](https://developer.android.com/training/data-storage/shared-preferences) on learning about Shared Preferences mentionned it :

> If you're using the SharedPreferences API to save app settings, you should instead use getDefaultSharedPreferences() to get the default shared preference file for your entire app.

But it's deprecated with API lvl 29 (see the link you provided). Do you recommend we still use it ?